specification/security_model.adoc

```diff
@@ -1,9 +1,195 @@
+At the same time, availability of those assigned TDIs is out of this model scope
+as e.g. the host VMM could remove them from the TVM at any time it sees fit. All
+TVM confidential data must be cleared and wiped before the host software stack
+can fully reclaim an assigned TDI.
+
+== Threat Model
+
+=== Assets
+
+The CoVE-IO security model aims at protecting the following assets:
+
+1. The TVM confidential data, which includes its confidential main memory, code
+   and state.
+2. Any assigned device I/O space that is mapped to the TVM address space,
+   also known as device MMIO.
+3. System memory.
 
 === Adversary Model
 
-=== Scope and Threat Model
+The CoVE-IO security model aims at protecting the above-described TVM assets
+from the following adversaries:
+
+* _Privileged host software adversary_: This includes host software components
+  executing in S and M mode like the host firmware, kernel, hypervisor, VMM,
+  etc. As the system resource owner but also the TVMs lifecycle manager, those
+  components can access and control all devices on the system.
+
+* _Unprivileged host software adversary_: (HS mode??)
+
+* _Privileged guest software adversary_: (Other TVM guest kernels)
+
+* _Device firmware adversary_: (Other system firmware, DMA into unassigned TVMs)
+
+* _Hardware adversary_: This includes adversaries that have unlimited physical
+  access to the system and its devices, and can use simple or advanced hardware
+  attacks such as bus interposers, voltage/clock glitchers, microscopes, power
+  usage analyzer, etc. in order to access or tamper with the protected assets.
+
+=== Threats
+
+==== CoVE-IO-T001
+
+.CoVE-IO-T001
+|===
+| Asset | Threat | Adversary | Scope | Result
+
+| TVM confidential data
+| Disclosure
+| Privileged host software
+| In scope
+| Host component reads TVM confidential data
+
+5+| **Description** - A privileged host software component programs a TDI that
+    is assigned to a TVM. By accessing the device MMIO space, the host component
+    can program direct memory access destination addresses to its own address
+    space. Device generated data intended to be copied to the TVM is accessed by
+    the host component instead.
+
+|===
+
+==== CoVE-IO-T002
+
+.CoVE-IO-T002
+|===
+| Asset | Threat | Adversary | Scope | Result
+
+| TVM confidential data
+| Tamper
+| Privileged host software
+| In scope
+| Host component writes to TVM confidential memory
+
+5+| **Description** - A privileged host software component programs a TDI that
+    is assigned to a TVM. By accessing the device MMIO space, the host component
+    can modify the programmed direct memory access destination addresses, and
+    redirect it to unintended parts of the TVM address space.
+
+|===
+
+==== CoVE-IO-T003
+
+.CoVE-IO-T003
+|===
+| Asset | Threat | Adversary | Scope | Result
+
+| TVM confidential data
+| Denial of service
+| Privileged host software
+| **Not** in scope
```

Comment:
I am not sure if we really need list this.
We said before that availability is our of scope.